game/src/edit/roads.rs

```diff
@@ -1,84 +1,461 @@
+        }
+
+        if self.current_lane.is_some()
+            && ctx.canvas.get_cursor_in_screen_space().is_none()
+            && ctx.normal_left_click()
+        {
+            self.current_lane = None;
+            self.main_panel =
+                make_main_panel(ctx, app, app.primary.map.get_r(self.r), self.current_lane);
+            self.highlight_selection =
+                highlight_current_selection(ctx, app, self.r, self.current_lane);
+        }
+
+        Transition::Keep
+    }
+
+    fn draw(&self, g: &mut GfxCtx, _: &App) {
+        g.redraw(&self.highlight_selection.1);
+        self.top_panel.draw(g);
+        self.main_panel.draw(g);
     }
-    lanes
 }
 
-fn lane_type_codes() -> Vec<(LaneType, char)> {
-    vec![
-        (LaneType::Driving, 'd'),
-        (LaneType::Parking, 'p'),
-        (LaneType::Sidewalk, 's'),
-        (LaneType::Shoulder, 'S'),
-        (LaneType::Biking, 'b'),
-        (LaneType::Bus, 't'), // transit
-        (LaneType::SharedLeftTurn, 'l'),
-        (LaneType::Construction, 'c'),
-        (LaneType::LightRail, 'r'),
+fn make_top_panel(
+    ctx: &mut EventCtx,
+    app: &App,
+    num_edit_cmds_originally: usize,
+    no_redo_cmds: bool,
+) -> Panel {
+    Panel::new(Widget::row(vec![
+        ctx.style()
+            .btn_solid_primary
+            .text("Finish")
+            .hotkey(Key::Escape)
+            .build_def(ctx),
+        ctx.style()
+            .btn_plain
+            .icon("system/assets/tools/undo.svg")
+            .disabled(app.primary.map.get_edits().commands.len() == num_edit_cmds_originally)
+            .hotkey(lctrl(Key::Z))
+            .build_widget(ctx, "undo"),
+        ctx.style()
+            .btn_plain
+            .icon("system/assets/tools/redo.svg")
+            .disabled(no_redo_cmds)
+            // TODO ctrl+shift+Z!
+            .hotkey(lctrl(Key::Y))
+            .build_widget(ctx, "redo"),
+    ]))
+    .aligned(HorizontalAlignment::Center, VerticalAlignment::Top)
+    .build(ctx)
+}
+
+fn make_main_panel(
+    ctx: &mut EventCtx,
+    app: &App,
+    road: &Road,
+    current_lane: Option<LaneID>,
+) -> Panel {
+    let map = &app.primary.map;
+
+    let modify_lane = if let Some(l) = current_lane {
+        let lane = map.get_l(l);
+        let idx = road.offset(l);
+
+        Widget::row(vec![
+            ctx.style()
+                .btn_plain
+                .icon("system/assets/tools/trash.svg")
+                .disabled(road.lanes_ltr().len() == 1)
+                .build_widget(ctx, "delete lane"),
+            ctx.style()
+                .btn_plain
+                .text("flip direction")
+                .disabled(!can_reverse(lane.lane_type))
+                .build_def(ctx),
+            Line("Width").secondary().into_widget(ctx),
+            Widget::dropdown(ctx, "width", lane.width, width_choices(app, l)),
+            ctx.style()
+                .btn_plain
+                .text("<")
```

Comment:
I worked hard to exorcise text "arrows" for their more handsome kin: `ctx.style().btn_prev()`

Reply:
Whoops, this is why code review really helps. Thanks!